test_lib/sla.py

```diff
@@ -125,119 +216,95 @@ def name(self, name):
         self._name = name
 
     @property
-    def attached_service_level_name(self):
-        return self._attached_service_level_name
-
-    @attached_service_level_name.setter
-    def attached_service_level_name(self, service_level_name):
-        self._attached_service_level_name = service_level_name
+    def attached_service_level(self):
+        return self._attached_service_level
 
     @property
-    def attached_service_level_shares(self):
-        return self._attached_service_level_shares
-
-    @attached_service_level_shares.setter
-    def attached_service_level_shares(self, service_level_shares):
-        self._attached_service_level_shares = service_level_shares
+    def attached_service_level_name(self):
+        return self._attached_service_level.name
 
     def attach_service_level(self, service_level):
-        query = 'ATTACH SERVICE_LEVEL {service_level_name} TO {role_name}' \
-            .format(service_level_name=service_level.name,
-                    role_name=self.name)
+        """
+        :param auth_name: it may be role name or user name
+        """
+        query = 'ATTACH SERVICE_LEVEL {service_level_name} TO {role_name}'\
+                .format(service_level_name=service_level.name,
+                        role_name=self.name)
         if self.verbose:
-            self.log.debug('Attach service level query: {}'.format(query))
+            logger.debug('Attach service level query: {}'.format(query))
         self.session.execute(query)
-        self.log.debug('Service level "{}" has been attached to {} role'.format(service_level.name, self.name))
-        self.attached_service_level_name = service_level.name
-        self.attached_service_level_shares = service_level.service_shares \
-            if service_level.service_shares \
-            else DEFAULT_SERVICE_LEVEL_SHARES
+        logger.debug('Service level "{}" has been attached to {} role'.format(service_level.name, self.name))
+        self._attached_service_level = service_level
 
     def detach_service_level(self):
-        query = 'DETACH SERVICE_LEVEL FROM {role_name}' \
-            .format(role_name=self.name)
+        """
+        :param auth_name: it may be role name or user name
+        """
+        query = 'DETACH SERVICE_LEVEL FROM {role_name}'\
+                .format(role_name=self.name)
         if self.verbose:
-            self.log.debug('Detach service level query: {}'.format(query))
+            logger.debug('Detach service level query: {}'.format(query))
```

Comment:
Please, use in all log messages lazy format

Reply:
Fixed

Reply:
It is not fixed

Reply:
Sorry, I missed some of them. Should be fixed now.